superbit_lensing/galsim/mock_superBIT_data.py

```diff
@@ -393,16 +401,37 @@ def make_a_star(ud, wcs, affine, optics, sbparams, logprint):
     # This is still an x/y corrdinate
     uv_pos = affine.toWorld(image_pos)
 
-    # Draw star flux at random; based on distribution of star fluxes in real images
-    flux_dist = galsim.DistDeviate(ud, function = lambda x:x**-1.5, x_min = 6, x_max = 47586)
-    star_flux=flux_dist()
-    if sbparams.bandpass=='crates_adu_b':
-        star_flux*=0.8271672
+    # Draw star flux at random; based on either a semi-analytic distribution or GAIA stars
+    # Default to blue stars, which are plenty bright
+
+    index = obj_index - 1
+
+    if star_cat is not None:
+        if sbparams.bandpass=='crates_adu_shape':
+            star_flux = star_cat['bit_flux_shape'][index]
+
+        elif sbparams.bandpass=='crates_adu_b':
+            star_flux = star_cat['bit_flux_b'][index]
+
+        else:
+            star_flux = star_cat['bit_flux_b'][index]
+
+        star_flux*=sbparams.exp_time
+
+
+    else:
+        pud = np.random.default_rng()
+        p = pud.power(0.4)
+        flux_p = (10/p) - 10.
+        star_flux = flux_p
+
+        if sbparams.bandpass=='crates_adu_b':
+            star_flux*=0.8271672
 
     # Generate PSF at location of star, convolve with optical model to make a star
     deltastar = galsim.DeltaFunction(flux=star_flux)
     jitter_psf = galsim.Gaussian(flux=1,fwhm=sbparams.jitter_fwhm)
-    star=galsim.Convolve([jitter_psf,deltastar,optics])
+    star=galsim.Convolve([jitter_psf,deltastar])
```

Comment:
Same as above - we need to keep `optics` in but the gauss branch can set it to `None` early on

Reply:
See second comment above: a boolean flag in the galsim config file could turn `optics` into a `galsim.DeltaFunction()` with `flux = 1`